hooks/_common.sh

```diff
@@ -170,6 +170,59 @@ function common::is_hook_run_on_whole_repo {
   fi
 }
 
+#######################################################################
+# Get the number of CPU logical cores available to pre-commit use
+# Arguments:
+#  parallelism_ci_cpu_cores (string) Used in corner cases when CPU cores
+#    can't be calculated
+# Outputs:
+#   Return number CPU logical cores, rounded to below integer
+#######################################################################
+function common::get_cpu_num {
+  local -r parallelism_ci_cpu_cores=$1
+
+  local millicpu
+
+  if [[ -f /sys/fs/cgroup/cpu/cpu.cfs_quota_us ]]; then
+    # Inside K8s pod or DInD in K8s
+    millicpu=$(cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us)
+
+    if [[ $millicpu -eq -1 ]]; then
+      # K8s no limits or in DinD
+      if [[ ! $parallelism_ci_cpu_cores ]]; then
+        common::colorify "yellow" "Unable to calculate available CPU cors.\n" \
+          "You in K8s pod without limits or in DinD without limits propagation.\n" \
+          "To avoid possible harm, parallelism disabled.\n" \
+          "If you'd like reenable it - set corresponding limits, or set next for current hook:\n" \
+          "  args:\n" \
+          "    - --hook-config=--parallelism-ci-cpu-cores=N\n" \
+          "where N is the number of CPU cores you providing to pre-commit."
+
+        return 1
+      fi
+
+      return "$parallelism_ci_cpu_cores"
+    fi
+
+    return $((millicpu / 1000))
+  fi
+
+  if [[ -f /sys/fs/cgroup/cpu.max ]]; then
+    # Inside Linux (Docker?) container
+    millicpu=$(cut -d' ' -f1 /sys/fs/cgroup/cpu.max)
+
+    if [[ "$millicpu" == "max" ]]; then
+      # No limits
+      return "$(nproc 2> /dev/null || echo 1)"
+    fi
+    return $((millicpu / 1000))
+  fi
+
+  # On host machine
+  # `nproc` - linux, `sysctl -n hw.ncpu` - macOS, `echo 1` - fallback
+  return "$(nproc 2> /dev/null || sysctl -n hw.ncpu 2> /dev/null || echo 1)"
```

Comment:
Decoupling if/else into independent blocks makes logic less clear IMHO:
```suggestion
  else
    # Running directly on host machine
    # `nproc` - linux, `sysctl -n hw.ncpu` - BSD/macOS, `echo 1` - fallback
    return "$(nproc 2> /dev/null || sysctl -n hw.ncpu 2> /dev/null || echo 1)"
  fi
```

Reply:
It's a programming pattern/recommendation to return ASAP and then don't get a "stair"-code

previously it was if-elif-else, but before push it I refactor it in more readable way

Reply:
Returning from `if/else` block has nothing to do with additional `elif` conditionals.
```shell
if [[ $var == something ]]; then
  echo something
  return
elif [[ $var == something_else ]]; then
  echo something else
  return
fi
```
is more readable and consistent rather than this:
```shell
if [[ $var == something ]]; then
  echo something
  return
fi
if [[ $var == something_else ]]; then
  echo something else
  return
fi
```

Reply:
Nope, as it makes your code more atomic and implicitly says that these `if` blocks can be rearranged without any harm

Reply:
Given they are about the same thing, I see no reason to decouple and re-arrange them. Moreover this "option" to re-arrange them (as in moving to arbitrary places or exchanging their order) might break the logic completely.
Though... up to you — you're the owner of this solution =)